fundamental-concepts/data-structures-stdlib/1_records.md

```diff
@@ -0,0 +1,87 @@
+One of the most basic data types in DAML is the record. Here's an example:
+
+<pre>
+data Address = Address {
+  street: Text,
+  city: Text,
+  country: Text
+}
+</pre>
+
+You define a record by listing its field names together with their types. Records can be nested:
+
+<pre>
+data Person = Person {
+  name: Text,
+  address: Address,
+  age: Int
+}
+</pre>
+
+You can create a new `Person` record by assigning each field with a value:
+
+<pre>
+testPerson: Person
+testPerson = Person {name = "Alice", address = Address {street = "RabbitStreet", city = "QueenOfHearts", country = "Wonderland"}, age = "7"}
+</pre>
+
+When you write a function and you need to **deconstruct** a record you can **pattern match**:
+
+<pre>
+prettyAddress: Address -> Text
+prettyAddress (Address aStreet aCity aCompany) = aStreet <> " - " <> aCity <> " - " <> aCompany
+</pre>
+
+Here we bound the variables `aStreet`, `aCity` and `aCompany` to the fields of the given `Address`
+record in order of its definition.
+
+You can also use the `..` syntax to get the record field names bound to variables of the same name:
+
+<pre>
+prettyAddress: Address -> Text
+prettyAddress Address {..} = street <> " - " <> city <> " - " <> country
+</pre>
+
+Or you can simply use the field accessors:
+
+<pre>
+prettyAddress: Address -> Text
+prettyAddress addr = (street addr) <> " - " <> (city addr) <> " - " <> (country addr)
+</pre>
+
+>> What will the following return: age testPerson? <<
+=== 7
+
+>> What will the following return: prettyAddress $ address testPerson? <<
+===  RabbitStreet - QueenOfHearts - Wonderland
+
+You can think of a contract **template** as a record data type annotated with what **parties** are
+**signatories**, what parties are **observing** the contract and what **choices** to build
+transactions are made available by the contract. We will cover these concepts in depth in the
+following Katacoda's of the [fundamental concepts
+course](https://daml.com/learn/fundamental-concepts). For example, we can turn our `Person` data
+type into a simple contract template:
+
+<pre>
+template Person 
+  with
+    name: Party,
+    address: Address,
+    age: Int
+  where
+    signatory name
+</pre>
```

Comment:
Would suggest that we allow the user to copy/paste this code in an actual file, together with the scenario, and then answer the questions

Reply:
this is the case now.